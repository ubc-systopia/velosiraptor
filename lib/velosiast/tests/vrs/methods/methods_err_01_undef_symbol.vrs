// undefined varaible
fn foo(foo: int, bar: int) -> int
{
    foo + BAR
}

// undefined variable
fn foo(foo: int, bar: int) -> int
{
    foobar
}

// wrong type
fn foo(foo: int, bar: int) -> int
{
    foo()
}

// recursive call doesn't work
fn bar(foo: int) -> int
{
    bar()
}