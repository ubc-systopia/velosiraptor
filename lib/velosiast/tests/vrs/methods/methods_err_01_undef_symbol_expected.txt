error: no symbol named `BAR` found in the current scope
     --> tests/vrs/methods/methods_err_01_undef_symbol.vrs:4:11
      |
    4 |      foo + BAR
      |            ^^^ not found in this scope
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: no symbol named `foobar` found in the current scope
     --> tests/vrs/methods/methods_err_01_undef_symbol.vrs:10:5
      |
   10 |      foobar
      |      ^^^^^^ not found in this scope
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: symbol named `foo` was defined with a different kind
     --> tests/vrs/methods/methods_err_01_undef_symbol.vrs:16:5
      |
   16 |      foo()
      |      ^^^^^ this is the symbol with the wrong kind
     --> tests/vrs/methods/methods_err_01_undef_symbol.vrs:14:8
      |
   14 |  fn foo(foo: int, bar: int) -> int
      |         ^^^^^^^^ symbol was defined here
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: no symbol named `bar` found in the current scope
     --> tests/vrs/methods/methods_err_01_undef_symbol.vrs:22:5
      |
   22 |      bar()
      |      ^^^^^ not found in this scope
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

